erpnext/controllers/accounts_controller.py

```diff
@@ -910,10 +910,16 @@ def get_value_in_transaction_currency(self, account_currency, args, field):
 			return flt(args.get(field, 0) / self.get("conversion_rate", 1))
 
 	def validate_qty_is_not_zero(self):
-		if self.doctype != "Purchase Receipt":
-			for item in self.items:
-				if not item.qty:
-					frappe.throw(_("Item quantity can not be zero"))
+		for item in self.items:
+			if self.doctype == "Purchase Receipt" and item.rejected_qty:
+				continue
+			if not item.qty:
+				frappe.throw(
+					_("Row # {0}: Quantity for Item {1} cannot be zero").format(
+						item.idx, frappe.bold(item.item_code)
```

Comment:
Note to the reviewer:

I’m consolidating at least the message with the other near-duplicate (around line 3000, within `update_child_qty_rate()`):
```
	def validate_quantity(child_item, new_data):
		if not flt(new_data.get("qty")):
			frappe.throw(
				_("Row # {0}: Quantity for Item {1} cannot be zero").format(
					new_data.get("idx"), frappe.bold(new_data.get("item_code"))
				),
				title=_("Invalid Qty"),
			)
```

We may look at how `update_child_qty_rate()`, which is only used in tests, can be consolidated as well, but this should probably be a followup.